modules/data-mover-for-csi-snapshots.adoc

```diff
@@ -0,0 +1,197 @@
++
+[NOTE]
+====
+The VolSync Operator is required only for use with the Technology Preview Data Mover. The Operator is not required for using OADP GA features.
+====
+
+* Install the link:https://github.com/openshift/oadp-operator/blob/master/docs/install_olm.md[OADP Operator] using OLM.
+
+* Install the link:https://volsync.readthedocs.io/en/stable/installation/index.html[VolSync operator] using OLM.
+
+.Procedure
+
+. Configure a link:https://volsync.readthedocs.io/en/stable/usage/restic/index.html#id2[Restic secret].
++
+[source,yaml]
+----
+$ cat << EOF > ./restic-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <secret-name>
+type: Opaque
+stringData:
+  # The repository encryption key
+  RESTIC_PASSWORD: <secure-restic-password>
+EOF
+----
+
+. Create a Data Protection Application (DPA) similar to the following example. Note that `defaultPlugins` includes CSI.
++
+[source,yaml]
+----
+apiVersion: oadp.openshift.io/v1alpha1
+kind: DataProtectionApplication
+metadata:
+  name: velero-sample
+  namespace: openshift-adp
+spec:
+  features:
+    enableDataMover: true
+    dataMoverCredential: <secret-name>  <1>
+  backupLocations:
+    - velero:
+        config:
+          profile: default
+          region: us-east-1
+        credential:
+          key: cloud
+          name: cloud-credentials
+        default: true
+        objectStorage:
+          bucket: <bucket-name>
+          prefix: <bucket-prefix>
+        provider: aws
+  configuration:
+    restic:
+      enable: <true or false>
+    velero:
+      defaultPlugins:
+        - openshift
+        - aws
+        - csi
+      featureFlags:
+        - EnableCSI
+----
+<1> Add the Restic secret name from the previous step. If this is not done,  the default secret name `dm-credential` is used.
++
+The OADP operator installs two VolumeSnapshotMover custom resource definitions (CRDs), `VolumeSnapshotBackup` and `VolumeSnapshotRestore`.
++
+.Example `VolumeSnapshotBackup` CR
++
+[source,yaml]
+----
+apiVersion: datamover.oadp.openshift.io/v1alpha1
+kind: VolumeSnapshotBackup
+metadata:
+  name: <vsb-name>
+spec:
+  volumeSnapshotContent:
+    name: <snapcontent-name>
+  protectedNamespace: <adp-namespace>
+  resticSecretRef:
+    name: <restic-secret-name>
+----
+
++
+.Example `VolumeSnapshotRestore` CR
++
+[source,yaml]
+----
+apiVersion: datamover.oadp.openshift.io/v1alpha1
+kind: VolumeSnapshotRestore
+metadata:
+  name: <vsr-name>
+spec:
+  protectedNamespace: <protected-ns>
+  resticSecretRef:
+    name: <restic-secret-name>
+  volumeSnapshotMoverBackupRef:
+    sourcePVCData:
+      name: <source-pvc-name>
+      size: <source-pvc-size>
+    resticrepository: <your-restic-repo>
+    volumeSnapshotClassName: <vsclass-name>
+----
+
+. To back up a snapshot, do the following:
+
+.. Create a backup CR:
+
++
+[source,yaml]
+----
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: <backup-name>
+  namespace: <protected-ns>
```

Comment:
protected-ns should be where the operator is installed (`openshift-adp` by default), maybe add a point explaining that

Reply:
Done